src/DeviceHardDriveType.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+class DeviceHardDriveType extends CommonDeviceType
+{
+    public $additional_fields_for_dictionnary = ['manufacturer'];
+
+
+    public static function getTypeName($nb = 0)
+    {
+        return _n('Device hard drive type', 'Device hard drive types', $nb);
+    }
+
+        /**
+     * @since 0.84
+     *
+     * @param $itemtype
+     * @param $base               HTMLTableBase object
+     * @param $super              HTMLTableSuperHeader object (default NULL)
+     * @param $father             HTMLTableHeader object (default NULL)
+     * @param $options   array
+     **/
+    public static function getHTMLTableHeader(
+        $itemtype,
+        HTMLTableBase $base,
+        HTMLTableSuperHeader $super = null,
+        HTMLTableHeader $father = null,
```

Comment:
If the default value is null then the type must probably be nullable to avoid errors
```suggestion
        ?HTMLTableSuperHeader $super = null,
        ?HTMLTableHeader $father = null,
```